lib/plugins/package/lib/zipService.js

```diff
@@ -142,47 +150,24 @@ function excludeNodeDevDependencies(servicePath) {
       return !isNodeModulesDir;
     });
 
-    return BbPromise.map(packageJsonPaths, (packageJsonPath) => {
+    // NOTE: using mapSeries here for a sequential computation (w/o race conditions)
+    return BbPromise.mapSeries(packageJsonPaths, (packageJsonPath) => {
       // the path where the package.json file lives
       const fullPath = path.join(servicePath, packageJsonPath);
       const dirWithPackageJson = fullPath.replace(path.join(path.sep, 'package.json'), '');
 
-      const childProc = childProcess.spawn('npm',
-        ['ls', '--dev=true', '--parseable=true', '--silent'],
+      return childProcess.execAsync(
+        `npm ls --dev=true --parseable=true --silent >> ${nodeDevDepFile}`,
```

Comment:
If you append to `${nodeDevDepFile}` then you should make sure to delete it before the `BbPromise.mapSeries()` above . Otherwise it might have some surprises in store for you. Why not put it in the artifacts folder (i.e. `.serverless` or whatever is provided in the `--package` command line) though? I think that folder is cleaned and recreated on every packaging?